### PR TITLE
Using custom branch name for auto backporting PRs

### LINKED ISSUE #1712: [BUG] Auto-backport is failing: permission denied
**Describe the bug**

```
remote: Permission to opensearch-project/OpenSearch.git denied to github-actions[bot].
```

**To Reproduce**
1. Label an issue backport 1.x.
2. See failed job.

https://github.com/opensearch-project/OpenSearch/runs/4509880595?check_suite_focus=true

### PR DESCRIPTION
**Pull request by VachaShah**
Signed-off-by: Vacha Shah <vachshah@amazon.com>

### Description
Fixes the issue for custom branch name for auto backport workflow. This will help in defining permissions properly for branches. For now, I have added the custom branch name changes in my [fork](https://github.com/VachaShah/backport) of the GHA repo, hence I am using a release version from my forked repo. Once the concerned change is merged to the backport GHA repo [tibdex/backport](https://github.com/tibdex/backport), I will update the GHA repo name.

After this PR is merged, I will verify it by running the backport workflow on a PR.
 
### Issues Resolved
#1712 
 
### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by saratvemulapalli** [comment]
start gradle check

**Review by saratvemulapalli — APPROVED** [review]
Nice work @VachaShah !

**Review by CEHENKLE — APPROVED** [review]

**Comment by CEHENKLE** [comment]
start gradle check

**Comment by saratvemulapalli** [comment]
> ❌   Gradle Check failure [1f03322](https://github.com/opensearch-project/OpenSearch/commit/1f033228ccfbaf8d86f8ff2397d2657e6966ac85) [Log 1793](https://ci.opensearch.org/logs/ci/workflow/OpenSearch_CI/PR_Checks/Gradle_Check/gradle_check_1793.log)
> 
> [Reports 1793](https://ci.opensearch.org/logs/ci/workflow/OpenSearch_CI/PR_Checks/Gradle_Check/gradle_check_1793_reports.zip)

`:qa:mixed-cluster:v1.3.0#mixedClusterTest` is failing.

**Comment by saratvemulapalli** [comment]
start gradle check

File: .github/workflows/backport.yml
Comment (dblock) [inline .github/workflows/backport.yml:20]: Is this going to create stale branches every time a PR is updated that has a backport label?

**Comment by dblock** [comment]
I recommend PRing the changes in VachaShah/backport first to at least get an idea of whether that change will be upstreamed, or get feedback on it, before we start using it everywhere.

File: .github/workflows/backport.yml
Comment (VachaShah) [reply .github/workflows/backport.yml:20]: A branch will only be created once the PR is merged. So if a PR number 100 is to be backported to release branch 1.x, the branch that will be created in the process is `backport/backport-100-to-1.x` after the PR is merged.

**Comment by VachaShah** [comment]
> I recommend PRing the changes in VachaShah/backport first to at least get an idea of whether that change will be upstreamed, or get feedback on it, before we start using it everywhere.

Do you mean opening a PR on my fork VachaShah/backport or on the original GHA repo tibdex/backport? For tibdex/backport, I have opened a PR https://github.com/tibdex/backport/pull/81 for this change. I don't see a lot of activity on it currently, so I figured I will add it to my fork VachaShah/backport and tested it on my OpenSearch forked repo.

**Comment by dblock** [comment]
> > I recommend PRing the changes in VachaShah/backport first to at least get an idea of whether that change will be upstreamed, or get feedback on it, before we start using it everywhere.
> 
> Do you mean opening a PR on my fork VachaShah/backport or on the original GHA repo tibdex/backport? For tibdex/backport, I have opened a PR [tibdex/backport#81](https://github.com/tibdex/backport/pull/81) for this change. I don't see a lot of activity on it currently, so I figured I will add it to my fork VachaShah/backport and tested it on my OpenSearch forked repo.

Perfect. I meant that I would wait a little before trying to use the build from your fork.

**Comment by VachaShah** [comment]
Hi @dblock! I haven't received a review yet on the PR in tibdex/backport, I was wondering should we continue to wait?

**Comment by dblock** [comment]
> Hi @dblock! I haven't received a review yet on the PR in tibdex/backport, I was wondering should we continue to wait?

Nah, after a week I think a private build off a fork makes sense.

**Comment by VachaShah** [comment]
> > Hi @dblock! I haven't received a review yet on the PR in tibdex/backport, I was wondering should we continue to wait?
> 
> Nah, after a week I think a private build off a fork makes sense.

Sounds good! I will merge this PR once the checks pass. Thanks @dblock!

**Comment by VachaShah** [comment]
start gradle check

**Pull request merged**
